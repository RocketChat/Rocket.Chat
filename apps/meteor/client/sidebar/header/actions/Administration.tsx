import { Sidebar } from '@rocket.chat/fuselage';
import { GenericMenu } from '@rocket.chat/ui-client';
import type { HTMLAttributes } from 'react';
import { useTranslation } from 'react-i18next';

import { useAdministrationMenu } from './hooks/useAdministrationMenu';

type AdministrationProps = Omit<HTMLAttributes<HTMLElement>, 'is'>;

const Administration = (props: AdministrationProps) => {
	const { t } = useTranslation();

	const sections = useAdministrationMenu();

	return <GenericMenu sections={sections} title={t('Administration')} is={Sidebar.TopBar.Action} {...props} />;
};

export default Administration;
