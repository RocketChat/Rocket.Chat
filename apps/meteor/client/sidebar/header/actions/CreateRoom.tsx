import { Sidebar } from '@rocket.chat/fuselage';
import { GenericMenu } from '@rocket.chat/ui-client';
import type { HTMLAttributes } from 'react';
import { useTranslation } from 'react-i18next';

import { useCreateRoom } from './hooks/useCreateRoomMenu';

type CreateRoomProps = Omit<HTMLAttributes<HTMLElement>, 'is'>;

const CreateRoom = (props: CreateRoomProps) => {
	const { t } = useTranslation();

	const sections = useCreateRoom();

	return <GenericMenu icon='edit-rounded' sections={sections} title={t('Create_new')} is={Sidebar.TopBar.Action} {...props} />;
};

export default CreateRoom;
