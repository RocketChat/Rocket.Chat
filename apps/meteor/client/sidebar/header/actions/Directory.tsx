import { Sidebar } from '@rocket.chat/fuselage';
import { useEffectEvent } from '@rocket.chat/fuselage-hooks';
import { useLayout, useRouter, useCurrentRoutePath } from '@rocket.chat/ui-contexts';
import type { HTMLAttributes } from 'react';

type DirectoryProps = Omit<HTMLAttributes<HTMLElement>, 'is'>;

const Directory = (props: DirectoryProps) => {
	const router = useRouter();
	const { sidebar } = useLayout();
	const handleDirectory = useEffectEvent(() => {
		sidebar.toggle();
		router.navigate('/directory');
	});
	const currentRoute = useCurrentRoutePath();

	return (
		<Sidebar.TopBar.Action {...props} icon='notebook-hashtag' onClick={handleDirectory} pressed={currentRoute?.includes('/directory')} />
	);
};

export default Directory;
