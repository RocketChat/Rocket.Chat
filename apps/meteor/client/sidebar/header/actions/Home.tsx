import { Sidebar } from '@rocket.chat/fuselage';
import { useEffectEvent } from '@rocket.chat/fuselage-hooks';
import { useRouter, useLayout, useSetting, useCurrentRoutePath } from '@rocket.chat/ui-contexts';
import type { HTMLAttributes } from 'react';

type SidebarHeaderActionHomeProps = Omit<HTMLAttributes<HTMLElement>, 'is'>;

const SidebarHeaderActionHome = (props: SidebarHeaderActionHomeProps) => {
	const router = useRouter();
	const { sidebar } = useLayout();
	const showHome = useSetting('Layout_Show_Home_Button');
	const handleHome = useEffectEvent(() => {
		sidebar.toggle();
		router.navigate('/home');
	});
	const currentRoute = useCurrentRoutePath();

	return showHome ? <Sidebar.TopBar.Action {...props} icon='home' onClick={handleHome} pressed={currentRoute?.includes('/home')} /> : null;
};

export default SidebarHeaderActionHome;
