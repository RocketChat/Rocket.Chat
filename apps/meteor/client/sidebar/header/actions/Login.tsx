import { Sidebar } from '@rocket.chat/fuselage';
import { useSessionDispatch } from '@rocket.chat/ui-contexts';
import type { HTMLAttributes } from 'react';
import { useTranslation } from 'react-i18next';

type LoginProps = Omit<HTMLAttributes<HTMLElement>, 'is'>;

const Login = (props: LoginProps) => {
	const setForceLogin = useSessionDispatch('forceLogin');
	const { t } = useTranslation();

	return (
		<Sidebar.TopBar.Action
			{...props}
			primary
			icon='login'
			title={t('Sign_in_to_start_talking')}
			onClick={(): void => setForceLogin(true)}
		/>
	);
};

export default Login;
