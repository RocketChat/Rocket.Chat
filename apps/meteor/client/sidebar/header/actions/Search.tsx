import { Sidebar } from '@rocket.chat/fuselage';
import { useEffectEvent, useOutsideClick } from '@rocket.chat/fuselage-hooks';
import type { HTMLAttributes } from 'react';
import { useState, useEffect, useRef } from 'react';
import tinykeys from 'tinykeys';

import SearchList from '../../search/SearchList';

type SearchProps = Omit<HTMLAttributes<HTMLElement>, 'is'>;

const Search = (props: SearchProps) => {
	const [searchOpen, setSearchOpen] = useState(false);

	const ref = useRef<HTMLElement>(null);
	const handleCloseSearch = useEffectEvent(() => {
		setSearchOpen(false);
	});

	useOutsideClick([ref], handleCloseSearch);

	const openSearch = useEffectEvent(() => {
		setSearchOpen(true);
	});

	useEffect(() => {
		const unsubscribe = tinykeys(window, {
			'$mod+K': (event) => {
				event.preventDefault();
				openSearch();
			},
			'$mod+P': (event) => {
				event.preventDefault();
				openSearch();
			},
		});

		return (): void => {
			unsubscribe();
		};
	}, [openSearch]);

	return (
		<>
			<Sidebar.TopBar.Action icon='magnifier' onClick={openSearch} {...props} />
			{searchOpen && <SearchList ref={ref} onClose={handleCloseSearch} />}
		</>
	);
};

export default Search;
