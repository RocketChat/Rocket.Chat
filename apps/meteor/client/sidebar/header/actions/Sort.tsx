import { Sidebar } from '@rocket.chat/fuselage';
import { GenericMenu } from '@rocket.chat/ui-client';
import { useLayout } from '@rocket.chat/ui-contexts';
import type { HTMLAttributes } from 'react';
import { useTranslation } from 'react-i18next';

import { useSortMenu } from './hooks/useSortMenu';

type SortProps = Omit<HTMLAttributes<HTMLElement>, 'is'>;

const Sort = (props: SortProps) => {
	const { t } = useTranslation();
	const { sidebar } = useLayout();
	const sections = useSortMenu();

	return (
		<GenericMenu
			icon='sort'
			sections={sections}
			title={t('Display')}
			selectionMode='multiple'
			is={Sidebar.TopBar.Action}
			disabled={sidebar.isCollapsed}
			{...props}
		/>
	);
};

export default Sort;
