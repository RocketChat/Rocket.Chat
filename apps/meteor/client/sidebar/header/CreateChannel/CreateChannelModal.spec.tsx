import CreateChannelModal from './CreateChannelModal';
import { testCreateChannelModal } from '../../../NavBarV2/NavBarPagesGroup/actions/testCreateChannelModal';

jest.mock('../../../lib/utils/goToRoomById', () => ({
	goToRoomById: jest.fn(),
}));

testCreateChannelModal(CreateChannelModal);
