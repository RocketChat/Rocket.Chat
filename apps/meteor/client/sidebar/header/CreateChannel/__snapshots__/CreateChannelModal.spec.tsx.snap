// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`renders Default without crashing 1`] = `
<body>
  <div>
    <dialog
      aria-labelledby=":r2o:-title"
      aria-modal="true"
      class="rcx-box rcx-box--full rcx-modal"
      data-qa="create-channel-modal"
      open=""
    >
      <form
        class="rcx-box rcx-box--full rcx-modal__inner rcx-css-1e2ego0"
        id=":r2o:"
      >
        <header
          class="rcx-box rcx-box--full rcx-modal__header"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__header-inner"
          >
            <h2
              class="rcx-box rcx-box--full rcx-modal__title rcx-css-trljwa rcx-css-lma364"
              id=":r2o:-title"
            >
              Create_channel
            </h2>
            <button
              aria-label="Close"
              class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button rcx-css-trljwa rcx-css-lma364"
              tabindex="-1"
              title="Close"
              type="button"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-cross rcx-icon rcx-css-4pvxx3"
              >
                
              </i>
            </button>
          </div>
        </header>
        <div
          class="rcx-box rcx-box--full rcx-modal__content rcx-css-1vw7itl"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__content-wrapper rcx-css-rh8v0f"
          >
            <fieldset
              class="rcx-box rcx-box--full rcx-field-group rcx-css-heem01"
              role="group"
            >
              <div
                class="rcx-box rcx-box--full rcx-field rcx-field-group__item"
              >
                <label
                  class="rcx-box rcx-box--full rcx-field__label rcx-label"
                  for=":r2p:"
                >
                  Name
                  <span
                    aria-hidden="true"
                    class="rcx-box rcx-box--full rcx-label__required rcx-css-b7g1a9"
                  >
                    *
                  </span>
                </label>
                <span
                  class="rcx-box rcx-box--full rcx-field__row"
                >
                  <label
                    class="rcx-box rcx-box--full rcx-label rcx-box rcx-box--full rcx-box--animated rcx-input-box__wrapper"
                  >
                    <input
                      aria-describedby=":r2p:-error :r2p:-hint"
                      aria-invalid="false"
                      aria-required="true"
                      class="rcx-box rcx-box--full rcx-box--animated rcx-input-box--undecorated rcx-input-box--type-text rcx-input-box"
                      data-qa-type="channel-name-input"
                      id=":r2p:"
                      name="name"
                      size="1"
                      type="text"
                    />
                    <span
                      class="rcx-box rcx-box--full rcx-input-box__addon"
                    >
                      <i
                        aria-hidden="true"
                        class="rcx-box rcx-box--full rcx-icon--name-hashtag-lock rcx-icon rcx-css-4pvxx3"
                      >
                        
                      </i>
                    </span>
                  </label>
                </span>
                <span
                  class="rcx-box rcx-box--full rcx-field__hint"
                  id=":r2p:-hint"
                >
                  No_spaces_or_special_characters
                </span>
              </div>
              <div
                class="rcx-box rcx-box--full rcx-field rcx-field-group__item"
              >
                <label
                  class="rcx-box rcx-box--full rcx-field__label rcx-label"
                  for=":r2q:"
                >
                  Topic
                </label>
                <span
                  class="rcx-box rcx-box--full rcx-field__row"
                >
                  <input
                    aria-describedby=":r2q:-hint"
                    class="rcx-box rcx-box--full rcx-box--animated rcx-input-box--type-text rcx-input-box"
                    data-qa-type="channel-topic-input"
                    id=":r2q:"
                    name="topic"
                    size="1"
                    type="text"
                  />
                </span>
                <span
                  class="rcx-box rcx-box--full rcx-field__hint"
                  id=":r2q:-hint"
                >
                  Displayed_next_to_name
                </span>
              </div>
              <div
                class="rcx-box rcx-box--full rcx-field rcx-field-group__item"
              >
                <label
                  class="rcx-box rcx-box--full rcx-field__label rcx-label"
                  for=":r30:"
                >
                  Members
                </label>
                <div
                  class="rcx-box rcx-box--full rcx-select"
                  data-qa="create-channel-users-autocomplete"
                  data-qa-type="user-auto-complete-input"
                  id=":r30:"
                >
                  <div
                    class="rcx-box rcx-box--full rcx-css-1sr8su7"
                  >
                    <div
                      class="rcx-box rcx-box--full rcx-css-w398ts"
                      role="listbox"
                    >
                      <input
                        aria-haspopup="listbox"
                        class="rcx-box rcx-box--full rcx-box--animated rcx-input-box--undecorated rcx-input-box rcx-css-trljwa rcx-css-pqpp8q"
                        placeholder="Add_people"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="rcx-box rcx-box--full rcx-select__addon rcx-css-x7bl3q rcx-css-1bprq55"
                  >
                    <i
                      aria-hidden="true"
                      class="rcx-box rcx-box--full rcx-icon--name-chevron-down rcx-icon rcx-css-4pvxx3"
                    >
                      
                    </i>
                  </div>
                </div>
              </div>
              <div
                class="rcx-box rcx-box--full rcx-field rcx-field-group__item"
              >
                <span
                  class="rcx-box rcx-box--full rcx-field__row"
                >
                  <label
                    class="rcx-box rcx-box--full rcx-field__label rcx-label"
                    for=":r2r:"
                  >
                    Private
                  </label>
                  <label
                    class="rcx-box rcx-box--full rcx-toggle-switch"
                  >
                    <input
                      aria-describedby=":r2r:-hint"
                      checked=""
                      class="rcx-box rcx-box--full rcx-toggle-switch__input"
                      id=":r2r:"
                      type="checkbox"
                    />
                    <i
                      aria-hidden="true"
                      class="rcx-box rcx-box--full rcx-toggle-switch__fake"
                    />
                  </label>
                </span>
                <span
                  class="rcx-box rcx-box--full rcx-field__hint"
                  id=":r2r:-hint"
                >
                  People_can_only_join_by_being_invited
                </span>
              </div>
            </fieldset>
            <div
              class="rcx-box rcx-box--full rcx-box--animated rcx-accordion"
            >
              <section
                class="rcx-box rcx-box--full rcx-accordion-item"
              >
                <div
                  aria-controls=":r32:"
                  aria-expanded="false"
                  class="rcx-box rcx-box--full rcx-box--animated rcx-accordion-item__bar"
                  role="button"
                  tabindex="0"
                >
                  <h2
                    class="rcx-box rcx-box--full rcx-accordion-item__title"
                    id=":r31:"
                  >
                    Advanced_settings
                  </h2>
                  <span
                    class="rcx-box rcx-box--full rcx-chevron"
                  >
                    <i
                      aria-hidden="true"
                      class="rcx-box rcx-box--full rcx-icon--name-chevron-down rcx-icon rcx-css-f8dsuo"
                    >
                      
                    </i>
                  </span>
                </div>
                <div
                  class="rcx-box rcx-box--full rcx-box--animated rcx-accordion-item__panel"
                  id=":r32:"
                >
                  <fieldset
                    class="rcx-box rcx-box--full rcx-field-group"
                    role="group"
                  >
                    <h5
                      class="rcx-box rcx-box--full rcx-field-group__item rcx-css-1rcvt7i"
                    >
                      Security_and_permissions
                    </h5>
                    <div
                      class="rcx-box rcx-box--full rcx-field rcx-field-group__item"
                    >
                      <span
                        class="rcx-box rcx-box--full rcx-field__row"
                      >
                        <label
                          class="rcx-box rcx-box--full rcx-field__label rcx-label"
                          for=":r2s:"
                        >
                          Federation_Matrix_Federated
                        </label>
                        <label
                          class="rcx-box rcx-box--full rcx-toggle-switch"
                        >
                          <input
                            aria-describedby=":r2s:-hint"
                            class="rcx-box rcx-box--full rcx-toggle-switch__input"
                            disabled=""
                            id=":r2s:"
                            type="checkbox"
                          />
                          <i
                            aria-hidden="true"
                            class="rcx-box rcx-box--full rcx-toggle-switch__fake"
                          />
                        </label>
                      </span>
                      <span
                        class="rcx-box rcx-box--full rcx-field__hint"
                        id=":r2s:-hint"
                      >
                        error-this-is-a-premium-feature
                      </span>
                    </div>
                    <div
                      class="rcx-box rcx-box--full rcx-field rcx-field-group__item"
                    >
                      <span
                        class="rcx-box rcx-box--full rcx-field__row"
                      >
                        <label
                          class="rcx-box rcx-box--full rcx-field__label rcx-label"
                          for=":r2u:"
                        >
                          Encrypted
                        </label>
                        <label
                          class="rcx-box rcx-box--full rcx-toggle-switch"
                        >
                          <input
                            aria-describedby=":r2u:-hint"
                            class="rcx-box rcx-box--full rcx-toggle-switch__input"
                            disabled=""
                            id=":r2u:"
                            type="checkbox"
                          />
                          <i
                            aria-hidden="true"
                            class="rcx-box rcx-box--full rcx-toggle-switch__fake"
                          />
                        </label>
                      </span>
                      <span
                        class="rcx-box rcx-box--full rcx-field__hint"
                        id=":r2u:-hint"
                      >
                        Not_available_for_this_workspace
                      </span>
                    </div>
                    <div
                      class="rcx-box rcx-box--full rcx-field rcx-field-group__item"
                    >
                      <span
                        class="rcx-box rcx-box--full rcx-field__row"
                      >
                        <label
                          class="rcx-box rcx-box--full rcx-field__label rcx-label"
                          for=":r2t:"
                        >
                          Read_only
                        </label>
                        <label
                          class="rcx-box rcx-box--full rcx-toggle-switch"
                        >
                          <input
                            aria-describedby=":r2t:-hint"
                            class="rcx-box rcx-box--full rcx-toggle-switch__input"
                            disabled=""
                            id=":r2t:"
                            type="checkbox"
                          />
                          <i
                            aria-hidden="true"
                            class="rcx-box rcx-box--full rcx-toggle-switch__fake"
                          />
                        </label>
                      </span>
                      <span
                        class="rcx-box rcx-box--full rcx-field__hint"
                        id=":r2t:-hint"
                      >
                        Anyone_can_send_new_messages
                      </span>
                    </div>
                    <div
                      class="rcx-box rcx-box--full rcx-field rcx-field-group__item"
                    >
                      <span
                        class="rcx-box rcx-box--full rcx-field__row"
                      >
                        <label
                          class="rcx-box rcx-box--full rcx-field__label rcx-label"
                          for=":r2v:"
                        >
                          Broadcast
                        </label>
                        <label
                          class="rcx-box rcx-box--full rcx-toggle-switch"
                        >
                          <input
                            class="rcx-box rcx-box--full rcx-toggle-switch__input"
                            id=":r2v:"
                            type="checkbox"
                          />
                          <i
                            aria-hidden="true"
                            class="rcx-box rcx-box--full rcx-toggle-switch__fake"
                          />
                        </label>
                      </span>
                    </div>
                  </fieldset>
                </div>
              </section>
            </div>
          </div>
        </div>
        <div
          class="rcx-box rcx-box--full rcx-modal__footer rcx-css-17mu816"
        >
          <div
            class="rcx-button-group rcx-button-group--align-end"
            role="group"
          >
            <button
              class="rcx-box rcx-box--full rcx-button rcx-button-group__item"
              type="button"
            >
              <span
                class="rcx-button--content"
              >
                Cancel
              </span>
            </button>
            <button
              class="rcx-box rcx-box--full rcx-button--primary rcx-button rcx-button-group__item"
              data-qa-type="create-channel-confirm-button"
              type="submit"
            >
              <span
                class="rcx-button--content"
              >
                Create
              </span>
            </button>
          </div>
        </div>
      </form>
    </dialog>
  </div>
</body>
`;

exports[`renders DefaultVersion2 without crashing 1`] = `
<body>
  <div>
    <dialog
      aria-labelledby=":r33:-title"
      aria-modal="true"
      class="rcx-box rcx-box--full rcx-modal"
      data-qa="create-channel-modal"
      open=""
    >
      <form
        class="rcx-box rcx-box--full rcx-modal__inner rcx-css-1e2ego0"
        id=":r33:"
      >
        <header
          class="rcx-box rcx-box--full rcx-modal__header"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__header-inner"
          >
            <h2
              class="rcx-box rcx-box--full rcx-modal__title rcx-css-trljwa rcx-css-lma364"
              id=":r33:-title"
            >
              Create_channel
            </h2>
            <button
              aria-label="Close"
              class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button rcx-css-trljwa rcx-css-lma364"
              tabindex="-1"
              title="Close"
              type="button"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-cross rcx-icon rcx-css-4pvxx3"
              >
                
              </i>
            </button>
          </div>
        </header>
        <div
          class="rcx-box rcx-box--full rcx-modal__content rcx-css-1vw7itl"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__content-wrapper rcx-css-rh8v0f"
          >
            <fieldset
              class="rcx-box rcx-box--full rcx-field-group rcx-css-heem01"
              role="group"
            >
              <div
                class="rcx-box rcx-box--full rcx-field rcx-field-group__item"
              >
                <label
                  class="rcx-box rcx-box--full rcx-field__label rcx-label"
                  for=":r34:"
                >
                  Name
                  <span
                    aria-hidden="true"
                    class="rcx-box rcx-box--full rcx-label__required rcx-css-b7g1a9"
                  >
                    *
                  </span>
                </label>
                <span
                  class="rcx-box rcx-box--full rcx-field__row"
                >
                  <label
                    class="rcx-box rcx-box--full rcx-label rcx-box rcx-box--full rcx-box--animated rcx-input-box__wrapper"
                  >
                    <input
                      aria-describedby=":r34:-error :r34:-hint"
                      aria-invalid="false"
                      aria-required="true"
                      class="rcx-box rcx-box--full rcx-box--animated rcx-input-box--undecorated rcx-input-box--type-text rcx-input-box"
                      data-qa-type="channel-name-input"
                      id=":r34:"
                      name="name"
                      size="1"
                      type="text"
                    />
                    <span
                      class="rcx-box rcx-box--full rcx-input-box__addon"
                    >
                      <i
                        aria-hidden="true"
                        class="rcx-box rcx-box--full rcx-icon--name-hashtag-lock rcx-icon rcx-css-4pvxx3"
                      >
                        
                      </i>
                    </span>
                  </label>
                </span>
                <span
                  class="rcx-box rcx-box--full rcx-field__hint"
                  id=":r34:-hint"
                >
                  No_spaces_or_special_characters
                </span>
              </div>
              <div
                class="rcx-box rcx-box--full rcx-field rcx-field-group__item"
              >
                <label
                  class="rcx-box rcx-box--full rcx-field__label rcx-label"
                  for=":r35:"
                >
                  Topic
                </label>
                <span
                  class="rcx-box rcx-box--full rcx-field__row"
                >
                  <input
                    aria-describedby=":r35:-hint"
                    class="rcx-box rcx-box--full rcx-box--animated rcx-input-box--type-text rcx-input-box"
                    data-qa-type="channel-topic-input"
                    id=":r35:"
                    name="topic"
                    size="1"
                    type="text"
                  />
                </span>
                <span
                  class="rcx-box rcx-box--full rcx-field__hint"
                  id=":r35:-hint"
                >
                  Displayed_next_to_name
                </span>
              </div>
              <div
                class="rcx-box rcx-box--full rcx-field rcx-field-group__item"
              >
                <label
                  class="rcx-box rcx-box--full rcx-field__label rcx-label"
                  for=":r3b:"
                >
                  Members
                </label>
                <div
                  class="rcx-box rcx-box--full rcx-select"
                  data-qa="create-channel-users-autocomplete"
                  data-qa-type="user-auto-complete-input"
                  id=":r3b:"
                >
                  <div
                    class="rcx-box rcx-box--full rcx-css-1sr8su7"
                  >
                    <div
                      class="rcx-box rcx-box--full rcx-css-w398ts"
                      role="listbox"
                    >
                      <input
                        aria-haspopup="listbox"
                        class="rcx-box rcx-box--full rcx-box--animated rcx-input-box--undecorated rcx-input-box rcx-css-trljwa rcx-css-pqpp8q"
                        placeholder="Add_people"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="rcx-box rcx-box--full rcx-select__addon rcx-css-x7bl3q rcx-css-1bprq55"
                  >
                    <i
                      aria-hidden="true"
                      class="rcx-box rcx-box--full rcx-icon--name-chevron-down rcx-icon rcx-css-4pvxx3"
                    >
                      
                    </i>
                  </div>
                </div>
              </div>
              <div
                class="rcx-box rcx-box--full rcx-field rcx-field-group__item"
              >
                <span
                  class="rcx-box rcx-box--full rcx-field__row"
                >
                  <label
                    class="rcx-box rcx-box--full rcx-field__label rcx-label"
                    for=":r36:"
                  >
                    Private
                  </label>
                  <label
                    class="rcx-box rcx-box--full rcx-toggle-switch"
                  >
                    <input
                      aria-describedby=":r36:-hint"
                      checked=""
                      class="rcx-box rcx-box--full rcx-toggle-switch__input"
                      id=":r36:"
                      type="checkbox"
                    />
                    <i
                      aria-hidden="true"
                      class="rcx-box rcx-box--full rcx-toggle-switch__fake"
                    />
                  </label>
                </span>
                <span
                  class="rcx-box rcx-box--full rcx-field__hint"
                  id=":r36:-hint"
                >
                  People_can_only_join_by_being_invited
                </span>
              </div>
            </fieldset>
            <div
              class="rcx-box rcx-box--full rcx-box--animated rcx-accordion"
            >
              <section
                class="rcx-box rcx-box--full rcx-accordion-item"
              >
                <div
                  aria-controls=":r3d:"
                  aria-expanded="false"
                  class="rcx-box rcx-box--full rcx-box--animated rcx-accordion-item__bar"
                  role="button"
                  tabindex="0"
                >
                  <h2
                    class="rcx-box rcx-box--full rcx-accordion-item__title"
                    id=":r3c:"
                  >
                    Advanced_settings
                  </h2>
                  <span
                    class="rcx-box rcx-box--full rcx-chevron"
                  >
                    <i
                      aria-hidden="true"
                      class="rcx-box rcx-box--full rcx-icon--name-chevron-down rcx-icon rcx-css-f8dsuo"
                    >
                      
                    </i>
                  </span>
                </div>
                <div
                  class="rcx-box rcx-box--full rcx-box--animated rcx-accordion-item__panel"
                  id=":r3d:"
                >
                  <fieldset
                    class="rcx-box rcx-box--full rcx-field-group"
                    role="group"
                  >
                    <h5
                      class="rcx-box rcx-box--full rcx-field-group__item rcx-css-1rcvt7i"
                    >
                      Security_and_permissions
                    </h5>
                    <div
                      class="rcx-box rcx-box--full rcx-field rcx-field-group__item"
                    >
                      <span
                        class="rcx-box rcx-box--full rcx-field__row"
                      >
                        <label
                          class="rcx-box rcx-box--full rcx-field__label rcx-label"
                          for=":r37:"
                          id=":r37:-label"
                        >
                          Federation_Matrix_Federated
                        </label>
                        <label
                          class="rcx-box rcx-box--full rcx-toggle-switch"
                        >
                          <input
                            aria-describedby=":r37:-hint"
                            aria-labelledby=":r37:-label"
                            class="rcx-box rcx-box--full rcx-toggle-switch__input"
                            disabled=""
                            id=":r37:"
                            type="checkbox"
                          />
                          <i
                            aria-hidden="true"
                            class="rcx-box rcx-box--full rcx-toggle-switch__fake"
                          />
                        </label>
                      </span>
                      <span
                        class="rcx-box rcx-box--full rcx-field__hint"
                        id=":r37:-hint"
                      >
                        error-this-is-a-premium-feature
                      </span>
                    </div>
                    <div
                      class="rcx-box rcx-box--full rcx-field rcx-field-group__item"
                    >
                      <span
                        class="rcx-box rcx-box--full rcx-field__row"
                      >
                        <label
                          class="rcx-box rcx-box--full rcx-field__label rcx-label"
                          for=":r39:"
                          id=":r39:-label"
                        >
                          Encrypted
                        </label>
                        <label
                          class="rcx-box rcx-box--full rcx-toggle-switch"
                        >
                          <input
                            aria-describedby=":r39:-hint"
                            aria-labelledby=":r39:-label"
                            class="rcx-box rcx-box--full rcx-toggle-switch__input"
                            disabled=""
                            id=":r39:"
                            type="checkbox"
                          />
                          <i
                            aria-hidden="true"
                            class="rcx-box rcx-box--full rcx-toggle-switch__fake"
                          />
                        </label>
                      </span>
                      <span
                        class="rcx-box rcx-box--full rcx-field__hint"
                        id=":r39:-hint"
                      >
                        Not_available_for_this_workspace
                      </span>
                    </div>
                    <div
                      class="rcx-box rcx-box--full rcx-field rcx-field-group__item"
                    >
                      <span
                        class="rcx-box rcx-box--full rcx-field__row"
                      >
                        <label
                          class="rcx-box rcx-box--full rcx-field__label rcx-label"
                          for=":r38:"
                          id=":r38:-label"
                        >
                          Read_only
                        </label>
                        <label
                          class="rcx-box rcx-box--full rcx-toggle-switch"
                        >
                          <input
                            aria-describedby=":r38:-hint"
                            aria-labelledby=":r38:-label"
                            class="rcx-box rcx-box--full rcx-toggle-switch__input"
                            disabled=""
                            id=":r38:"
                            type="checkbox"
                          />
                          <i
                            aria-hidden="true"
                            class="rcx-box rcx-box--full rcx-toggle-switch__fake"
                          />
                        </label>
                      </span>
                      <span
                        class="rcx-box rcx-box--full rcx-field__hint"
                        id=":r38:-hint"
                      >
                        Anyone_can_send_new_messages
                      </span>
                    </div>
                    <div
                      class="rcx-box rcx-box--full rcx-field rcx-field-group__item"
                    >
                      <span
                        class="rcx-box rcx-box--full rcx-field__row"
                      >
                        <label
                          class="rcx-box rcx-box--full rcx-field__label rcx-label"
                          for=":r3a:"
                          id=":r3a:-label"
                        >
                          Broadcast
                        </label>
                        <label
                          class="rcx-box rcx-box--full rcx-toggle-switch"
                        >
                          <input
                            aria-describedby=":r3a:-hint"
                            aria-labelledby=":r3a:-label"
                            class="rcx-box rcx-box--full rcx-toggle-switch__input"
                            id=":r3a:"
                            type="checkbox"
                          />
                          <i
                            aria-hidden="true"
                            class="rcx-box rcx-box--full rcx-toggle-switch__fake"
                          />
                        </label>
                      </span>
                    </div>
                  </fieldset>
                </div>
              </section>
            </div>
          </div>
        </div>
        <div
          class="rcx-box rcx-box--full rcx-modal__footer rcx-css-17mu816"
        >
          <div
            class="rcx-button-group rcx-button-group--align-end"
            role="group"
          >
            <button
              class="rcx-box rcx-box--full rcx-button rcx-button-group__item"
              type="button"
            >
              <span
                class="rcx-button--content"
              >
                Cancel
              </span>
            </button>
            <button
              class="rcx-box rcx-box--full rcx-button--primary rcx-button rcx-button-group__item"
              data-qa-type="create-channel-confirm-button"
              type="submit"
            >
              <span
                class="rcx-button--content"
              >
                Create
              </span>
            </button>
          </div>
        </div>
      </form>
    </dialog>
  </div>
</body>
`;
